projects/client/RabbitMQ.Client/src/client/impl/ConsumerWorkService.cs

```diff
@@ -68,11 +62,21 @@ public void AddWork(IModel model, Action fn)
 
         public void RegisterKey(IModel model)
         {
+            if (model.ChannelNumber == 0)
```

Comment:
Yes, I noticed that the current design causes even the `MainSession` model to be registered with the `ConsumerWorkService`, even though no consumers can be registered on it. Seemed like a reasonable optimization to skip those registrations.